### ISSUE TITLE
 Entitas 0.47.8 incompatible with ScriptableObject

### ISSUE THREAD

**Issue by roygear** [sschmid/Entitas-CSharp#564]
tested in Window 7, unity 2017.3

If project contain a ScriptableObject class like this:
```
using UnityEngine;
[Game]
public class Configs : ScriptableObject
{
    public GameObject gameObject;
}
```
below error will appear after execute Jenny gen 
```
Sequence contains no elements
```

and generator won't work

**Comment by chiuan** [comment]
why are you need to do this ???

**Comment by roygear** [comment]
@chiuan you can make global setting on Unity's inspector. This important to Game designer

**Comment by StormRene** [comment]
What do you think about: 
`
using UnityEngine;
[Game]
public class ConfigsComponent : IComponent
{
    public Configs value;
}
public Configs : ScriptableObject {
<....>
}
`
Just write a simple InitializeSystem to set your Component.

**Comment by KumoKairo** [comment]
This thing actually worked back in the day, we used to store settings components like that. Just mark a custom `Settings : ScriptableObject` class with `[Game]` and it should generate an appropriate component:
```
SettingsComponent : IComponent
{
    Settings value;
}
```

So I guess it's an issue with the new generator?

**Comment by AVAVT** [comment]
It took some time to separate Component and ScriptableObject, but it looks cleaner now so I have no complaint. Entitas' Attribute above non-IComponent looks confusing.

**Comment by sschmid** [comment]
Can reproduce, works with the default code generator, doesn't work with roslyn. Will investigate...

**Comment by sschmid** [comment]
As a workaround please manually create  the ConfigsComponent

```csahrp
[Game]
public sealed class ConfigsComponent : Entitas.IComponent {
    public Configs value;
}
```

**Comment by sschmid** [comment]
Ok, fixed. Will be in next release.

In case anyone is interested: Explanation:
The fix for #532 introduced that behaviour because all inherited members where inspected too to support subclassed components. The same functionality was applied for non-components. I fixed this to not inspect sublasses of non-components.

**Comment by sschmid** [comment]
Fixed in [Entitas 0.47.9](https://github.com/sschmid/Entitas-CSharp/releases/tag/0.47.9)
